data/functions.sql

```diff
@@ -687,3 +687,14 @@ BEGIN
   RETURN CASE WHEN val IS NULL THEN 'null' ELSE to_json(val) END;
 END;
 $$ LANGUAGE plpgsql IMMUTABLE;
+
+-- the list of values was taken from the first 4 pages of taginfo for building
+CREATE OR REPLACE FUNCTION mz_building_kind_detail(val TEXT)
+RETURNS TEXT AS $$
+DECLARE
+BEGIN
+  RETURN CASE WHEN val = 'yes' THEN NULL
+              WHEN val IN ('house', 'residential', 'garage', 'apartments', 'hut', 'industrial', 'roof', 'shed', 'detached', 'commercial', 'garages', 'terrace', 'school', 'retail', 'farm_auxiliary', 'construction', 'church', 'barn', 'greenhouse', 'service', 'cabin', 'manufacture', 'farm', 'warehouse', 'civic', 'office', 'collapsed', 'university', 'public', 'hangar', 'entrance', 'hospital', 'static_caravan', 'hotel', 'chapel', 'semidetached_house', 'damaged', 'kindergarten', 'transportation', 'train_station', 'dormitory', 'storage_tank', 'trullo', 'stable', 'ger', 'houseboat', 'shop', 'college', 'ruins', 'semi', 'bunker', 'pajaru', 'mosque', 'bungalow', 'slurry_tank', 'supermarket', 'transformer_tower', 'duplex', 'factory', 'cowshed', 'constructie', 'dam', 'agricultural', 'tank', 'shelter', 'dwelling_house', 'temple', 'tower', 'agricultural', 'tank', 'shelter', 'dwelling_house', 'temple', 'tower', 'silo', 'residence', 'pavilion', 'store', 'support', 'allotment_house', 'summer_cottage', 'mobile_home', 'administrative', 'kiosk', 'other', 'boathouse', 'glasshouse', 'storage') THEN val
```

Comment:
> I wonder if we should exclude the building=collapsed and building=ruins from inclusion as a kind:building, perhaps by excluding them from the definition of *is_building in the YAML?

If a building is **collapsed** it still has a "footprint" _but we should probably zero out it's height_? I'm thinking of Christchurch earthquake in New Zealand and the HOT work around mapping all those buildings – we'd still want them to show up.

For buildings marked **ruins**, it looks like there is a tiny bit of overlap between [building=ruins](http://wiki.openstreetmap.org/wiki/Tag:building%3Druins) and [historic=ruin](http://wiki.openstreetmap.org/wiki/Tag:historic%3Druins). But they are different concepts. If it were historic I'd expect a POI to be generated, but for these basic buildings I think we'd only generate a label placement and pass thru `ruins` as the `kind_detail`. Ruins can still have height, so I wouldn't mess with that value.

I can see only showing the label position for these at zoom 17+, but the features themselves should be visible like normal based on their area.